plugins/modules/packaging/language/ansible_galaxy_install.py

```diff
@@ -0,0 +1,318 @@
+    ansible_version = None
+    is_ansible29 = None
+
+    output_params = ('type', 'name', 'dest', 'requirements_file', 'force')
+    module = dict(
+        argument_spec=dict(
+            type=dict(type='str', choices=('collection', 'role', 'both'), required=True),
+            name=dict(type='str'),
+            requirements_file=dict(type='path'),
+            dest=dict(type='path'),
+            force=dict(type='bool', default=False),
+            ack_ansible29=dict(type='bool', default=False),
+        ),
+        mutually_exclusive=[('name', 'requirements_file')],
+        required_one_of=[('name', 'requirements_file')],
+        required_if=[('type', 'both', ['requirements_file'])],
+        supports_check_mode=False,
+    )
+
+    command = 'ansible-galaxy'
+    command_args_formats = dict(
+        type=dict(fmt=lambda v: [] if v == 'both' else [v]),
+        galaxy_cmd=dict(),
+        requirements_file=dict(fmt=('-r', '{0}'),),
+        dest=dict(fmt=('-p', '{0}'),),
+        force=dict(fmt="--force", style=ArgFormat.BOOLEAN),
+    )
+    force_lang = "en_US.UTF-8"
+    check_rc = True
+
+    def _get_ansible_galaxy_version(self):
+        ansible_galaxy = self.module.get_bin_path("ansible-galaxy", required=True)
+        dummy, out, dummy = self.module.run_command([ansible_galaxy, "--version"], check_rc=True)
+        line = out.splitlines()[0]
+        match = self._RE_GALAXY_VERSION.match(line)
+        if not match:
+            raise RuntimeError("Unable to determine ansible-galaxy version from: {0}".format(line))
+        version = match.group("version")
+        version = tuple(int(x) for x in version.split('.')[:3])
+        return version
+
+    def __init_module__(self):
+        self.ansible_version = self._get_ansible_galaxy_version()
+        self.is_ansible29 = self.ansible_version < (2, 10)
+        if self.is_ansible29:
+            self._RE_INSTALL_OUTPUT = re.compile(r"^(?:.*Installing '(?P<collection>\w+\.\w+):(?P<cversion>[\d\.]+)'.*"
+                                                 r'|- (?P<role>\w+\.\w+) \((?P<rversion>[\d\.]+)\)'
+                                                 r' was installed successfully)$')
+        else:
+            # Collection install output changed:
+            # ansible-base 2.10:  "coll.name (x.y.z)"
+            # ansible-core 2.11+: "coll.name:x.y.z"
+            self._RE_INSTALL_OUTPUT = re.compile(r'^(?:(?P<collection>\w+\.\w+)(?: \(|:)(?P<cversion>[\d\.]+)\)?'
+                                                 r'|- (?P<role>\w+\.\w+) \((?P<rversion>[\d\.]+)\))'
+                                                 r' was installed successfully$')
+
+    @staticmethod
+    def __process_output_list__(*args):
+        if "None of the provided paths were usable" in args[1]:
+            return []
+        return args[1].splitlines()
+
+    def __list_element__(self, _type, path_re, elem_re):
```

Comment:
What is the point in populating the class with artificial dunder methods — this just screams "CPython internals" which is misleading. Also, if you want private methods, you could just use a single leading underscore. Otherwise, it looks like a lot of meta-magic while it's not what's actually happening.

Reply:
well, I don't get that same reading from it as you do, but fair enough, changing the methods in this class